cloudinit/sources/DataSourceAzure.py

```diff
@@ -611,16 +609,19 @@ def exc_cb(msg, exception):
                     if self.imds_poll_counter == self.imds_logging_threshold:
                         # Reducing the logging frequency as we are polling IMDS
                         self.imds_logging_threshold *= 2
-                        LOG.debug("Call to IMDS with arguments %s failed "
-                                  "with status code %s after %s retries",
-                                  msg, exception.code, self.imds_poll_counter)
+                        report_diagnostic_event(
+                            "Call to IMDS with arguments %s failed "
+                            "with status code %s after %s retries" %
+                            (msg, exception.code, self.imds_poll_counter),
+                            LOG.warning)
```

Comment:
@anhvoms Is this diagnostic event still needed given that we have the exact same one below? I'm in favor of removing one of them.

Reply:
removed.